synapse/state/__init__.py

```diff
@@ -554,7 +554,10 @@ async def resolve_state_groups(
             if conflicted_state:
                 logger.info("Resolving conflicted state for %r", room_id)
                 with Measure(self.clock, "state._resolve_events"):
-                    new_state = await resolve_events_with_store(
+                    # mypy doesn't like that new_state was previously mutable,
+                    # and now isn't. We don't mutate past this point though, so
+                    # tell it to go away.
+                    new_state = await resolve_events_with_store(  # type: ignore
```

Comment:
If there's something better to do here let me know. I think we could do it with temporary variables, but it would make the code harder to follow?

Reply:
`cast` to `MutableStateMap` might be more preferable,

though I'd be concerned about taking something immutable, falsely declaring it as mutable and then passing it to a function taking the mutable variant (that function 'could' change in the future, if we're being general).

I'd say that you should be preferring to lock things down (take the prior mutable and say it is now immutable).

Not sure about the specific risk in this scenario though.

Reply:
That seems reasonable. I think it is OK to falsely declare it as mutable here, see a7f2152593f8661c6c82b43a1c4ef48696cbc0df.